### PR TITLE
semantic-ui: CommunitySelectionModal: Remove unused Footer component

### PR DESCRIPTION
**Pull request by sakshamarora1**
Fixes: https://github.com/inveniosoftware/invenio-communities/issues/1209

### CODE REVIEW

**Review by zzacharo — APPROVED** [review]

**Review by 0einstein0 — APPROVED** [review]
Reviwed with @sakshamarora1 👀

**Review by zzacharo — COMMENTED** [review]
There are more places of this being hardcoded, see for example [here](https://github.com/inveniosoftware/invenio-app-rdm/blob/35165ab5a0509dd7b7e83bad0af8e7c28b240297/invenio_app_rdm/users_ui/templates/semantic-ui/invenio_app_rdm/users/header.html#L33)

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/components/CommunitySelectionModal/CommunitySelectionFooter.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const CommunitySelectionFooter = () => {
           <p>
             <Trans>
               Did not find a community that fits you? Upload without a community or{" "}
-              <a href="/communities/new">create your own.</a>
+              <a href="/communities-new">create your own.</a>
```
Comment (zzacharo) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/components/CommunitySelectionModal/CommunitySelectionFooter.js:20]: also all these places should be changed to utilize the config rather the hardcoded values

Comment (sakshamarora1) [reply invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/components/CommunitySelectionModal/CommunitySelectionFooter.js:20]: It's a javascript file, I'm not sure if we are already injecting config?

**Pull request merged**